### ISSUE TITLE
"Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size" causing memory leak

### ISSUE THREAD

**Issue by Oats87** [rancher/rancher#15719]
**Rancher versions:**
rancher/server or rancher/rancher: 2.0.6 (verified on 2.0.8 as well)

**Infrastructure Stack versions:**
kubernetes (if applicable): 1.10.1

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
This has been reproduced on both single-node and HA rancher 2.0

**Description:**

When accessing a specific project page in a downstream cluster from the Rancher UI, the Rancher container logs start spitting out:

```
E0919 17:06:17.822139       1 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
2018/09/19 17:06:17 [DEBUG] closing watcher for configMap
```

and in the web UI when viewing the Console, can see logs that look like

```
subscribe.js:287 Project Subscribe disconnected (sockId=null)
socket.js:57 Socket connecting (id=16, url=wss://<rancher-server>/v3/projects/<project-id>/subscribe...)
socket.js:241 Socket opened (state=reconnecting, id=16)
subscribe.js:277 Project Subscribe connected (sockId=16) (after 1 try, 0.145 sec)
socket.js:186 Socket null closed
```

Currently the suspicion is a config map on the downstream cluster is causing this issue.

After this occurs for a long enough time, the rancher container memory usage will cause the container to crash/system to OOM-kill rancher

**Comment by Oats87** [comment]
[gh-15719.yaml.zip](https://github.com/rancher/rancher/files/2400023/gh-15719.yaml.zip)
I have found a set of configmaps that can reproduce this issue. When these configmaps are created on a cluster (i.e. `kubectl create -f gh-15719.yaml`), they will be added to the default namespace of the cluster. When you access the "Default" project page and monitor the browser's console, you should see the thrashing of the Subscribe/Reconnecting/Connected messages, and checking the logs you should see the spam of the Unable to decode an event. After enough time, the system will hang.

/cc @kinarashah @cjellick @cloudnautique @aemneina

**Comment by Oats87** [comment]
<img width="1920" alt="screen shot 2018-09-20 at 9 48 37 am" src="https://user-images.githubusercontent.com/30601846/45833804-75238080-bcba-11e8-9a98-5492cf4996f3.png">
Still reproducible on master

**Comment by cjellick** [comment]
@Oats87 - fix is available in rancher/rancher:master (well, will be in roughly 10 minutes after this message). Can you give it a go and confirm the memory leak is fixed for you?

**Comment by sangeethah** [comment]
Tested with latest build from master:

Had the cofigmaps (provided in the issue) created in Default project of a Digital Ocean cluster.
Access the "Default" project using UI.

This results in the following errors seen continously in logs:
```
E0925 17:52:40.019610       7 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 17:52:41.042190       7 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 17:52:41.963214       7 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 17:52:43.747679       7 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 17:52:44.323568       7 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 17:52:44.888823       7 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
```

There is lot of socket connections made from UI:

<img width="1329" alt="screen shot 2018-09-25 at 10 53 59 am" src="https://user-images.githubusercontent.com/4266958/46032772-5c97d980-c0b1-11e8-9944-86f894fcbb97.png">

Memory consumption of rancher server does not increase (when still in the project page).

When the same scenario was tested with 2.1.0-rc4 server , the memory consumption of rancher server steadily increases (when  still in the project page)  seen it climb from ~20 % - 65% within few minutes.

**Comment by Oats87** [comment]
Tested with Master

Utilized reproducer configmap and created the configmap on the cluster; upon utilizing Rancher to view the default project, there was no memory leak that occurred.

```
E0925 22:16:39.423876       5 streamwatcher.go:109] Unable to decode an event from the watch stream: tunnel disconnect
E0925 22:16:39.423883       5 streamwatcher.go:109] Unable to decode an event from the watch stream: tunnel disconnect
E0925 22:16:39.423889       5 streamwatcher.go:109] Unable to decode an event from the watch stream: tunnel disconnect
E0925 22:16:39.423895       5 streamwatcher.go:109] Unable to decode an event from the watch stream: tunnel disconnect
E0925 22:16:39.423900       5 streamwatcher.go:109] Unable to decode an event from the watch stream: tunnel disconnect
E0925 22:16:39.674013       5 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 22:16:39.949626       5 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
2018/09/25 22:16:43 [INFO] Handling backend connection request [c-q8tcd]
E0925 22:16:44.111844       5 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
E0925 22:16:44.653693       5 streamwatcher.go:109] Unable to decode an event from the watch stream: object to decode was longer than maximum allowed size
```

```
Tue Sep 25 15:21:18 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         943        5893           8         985        6571
Swap:          2043           0        2043
Tue Sep 25 15:21:33 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         943        5894           8         985        6572
Swap:          2043           0        2043
Tue Sep 25 15:21:48 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         944        5893           8         985        6571
Swap:          2043           0        2043
Tue Sep 25 15:22:03 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         943        5893           8         985        6571
Swap:          2043           0        2043
Tue Sep 25 15:22:18 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         942        5894           8         985        6572
Swap:          2043           0        2043
Tue Sep 25 15:22:33 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         941        5895           8         985        6573
Swap:          2043           0        2043
Tue Sep 25 15:22:48 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         942        5894           8         985        6572
Swap:          2043           0        2043
Tue Sep 25 15:23:03 PDT 2018
              total        used        free      shared  buff/cache   available
Mem:           7822         941        5895           8         985        6573
Swap:          2043           0        2043
```

**Comment by sangeethah** [comment]
@Oats87 Could you confirm that this issue is fixed when testing with v2.1.0-rc5?
